train_gpt2.cu

```diff
@@ -53,6 +54,7 @@ This reads & runs in fp32, B=4, T=64, LR=1e-4, val/sample never (200),
 #include "utils.h"
 // defines: tokenizer_init, tokenizer_decode, tokenizer_free
 #include "tokenizer.h"
+#undef FLT_MAX
```

Comment:
I don't get this one. There's no re-definition I can see

Reply:
in math.h
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\float.h

#define FLT_MANT_DIG     24                      // # of bits in mantissa
**#define FLT_MAX          3.402823466e+38F        // max value**
#define FLT_MAX_10_EXP   38                      // max decimal exponent
#define FLT_MAX_EXP      128                     // max binary exponent
#define FLT_MIN          1.175494351e-38F        // min normalized positive value


Around line 686 in train_gpt2.cu - those definitions aren't the same so the compiler errors it out.

```
    // not INF, so we don't get NaNs accidentally when subtracting two values.
    const float FLT_MAX = 340282346638528859811704183484516925440.0f; // to avoid including float.h
    float maxval = -FLT_MAX;
    float sumval = 0.0f;

```

Reply:
I'd rather just change the (local) varname here, e.g. make it flt_max or something. undef is not amazing because it is a global state kind of statement

Reply:
Got it - updating...

Reply:
I think the NoneType CI error is from a PR I just merged from @Chillee , which I assumed was bug-free, should have paid more attention :) I could revert it, ... let me see if i can figure out what the source is here

Reply:
Not a problem. Will check the CI to see when it's clear. The fix is ready.

Reply:
@rosslwheeler I pushed a fix to master, try now

Reply:
Okay - let's see how it does now.

Reply:
haha I was just running the benchmark script - I assumed that CI would catch any other issues 😅